src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -427,5 +457,51 @@ private async Task AssertAutomaticReviewModifier(ClaimsPrincipal user, ReviewMod
                 throw new AuthorizationFailedException();
             }
         }
+
+        private async Task ApproveOrOutdateReviews(ClaimsPrincipal user, ReviewModel review, string revisionId)
+        {
+            //This method is called when a review is approved
+            var revisionModel = review.Revisions.Single(r => r.RevisionId == revisionId);
+            var revisionFile = revisionModel.Files.FirstOrDefault();
+            var codeFile = await _codeFileRepository.GetCodeFileAsync(revisionModel);
+
+            // Get reviews and mark them as approved if it matches current one
+            var pendingReviews = await _reviewsRepository.GetReviewsAsync(revisionFile.Language, revisionFile.PackageName, null, false);
+            foreach ( var r in pendingReviews)
+            {
+                // If review matches then mark it as approved
+                bool isReviewSame = await IsReviewSame(r, codeFile);
+                if (isReviewSame)
+                {
+                    var userId = user.GetGitHubLogin();
+                    r.Revisions.Last().Approvers.Add(userId);
+                }
+                else
+                {
+                    r.Revisions.Last().IsOutdated = true;
+                }
+                await _reviewsRepository.UpsertReviewAsync(r);
+            }
+        }
+
+        private async Task<ReviewModel> FindMatchingApprovedReview(ReviewModel review)
+        {
+            var revisionModel = review.Revisions.Last();
+            var revisionFile = revisionModel.Files.FirstOrDefault();
+            var codeFile = await _codeFileRepository.GetCodeFileAsync(revisionModel);
+
+            // Get automatic reviews and mark them as approved if it matches current one
+            var reviews = await _reviewsRepository.GetReviewsAsync(revisionFile.Language, revisionFile.PackageName, false, true);
+            foreach (var r in reviews)
+            {
+                // If review matches then mark it as approved
+                bool isReviewSame = await IsReviewSame(r, codeFile);
```

Comment:
Do we care if the approved revision is not last?

Reply:
In most cases, It shouldn't be. But I can think of a scenario where it breaks if we don't have API matching logic in both ways when I thought more about this comment. If someone creates a new revision as soon as last one is approved then it won't match. But this scenario is already handled by matching logic in both directions so system finds matching automatic review when manual review is approved so checking last revision is good enough.

Reply:
I have made a change to look for last approved revision and compare against it.